plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -0,0 +1,452 @@
+                            this,
+                            session,
+                            table.getSchemaName(),
+                            table.getTableName(),
+                            Optional.empty(),
+                            Optional.empty());
+                    return new BaseTable(operations, quotedTableName(table)).operations().current();
+                });
+
+        return getIcebergTableWithMetadata(
+                this,
+                tableOperationsProvider,
+                session,
+                table,
+                metadata);
+    }
+
+    @Override
+    public void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
+    {
+        BaseTable table = (BaseTable) loadTable(session, schemaTableName);
+        validateTableCanBeDropped(table, schemaTableName);
+        try {
+            deleteTable(schemaTableName.getSchemaName(), schemaTableName.getTableName());
+        }
+        catch (AmazonServiceException e) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, e);
+        }
+        dropTableData(table.io(), table.operations().current());
+        deleteTableDirectory(session, schemaTableName, hdfsEnvironment, new Path(table.location()));
+    }
+
+    @Override
+    public Transaction newCreateTableTransaction(
+            ConnectorSession session,
+            SchemaTableName schemaTableName,
+            Schema schema,
+            PartitionSpec partitionSpec,
+            String location,
+            Map<String, String> properties)
+    {
+        return newCreateTableTransaction(
+                session,
+                schemaTableName,
+                schema,
+                partitionSpec,
+                location,
+                properties,
+                Optional.of(session.getUser()));
+    }
+
+    @Override
+    public void renameTable(ConnectorSession session, SchemaTableName from, SchemaTableName to)
+    {
+        boolean newTableCreated = false;
+        try {
+            GetTableRequest getTableRequest = new GetTableRequest()
+                    .withDatabaseName(from.getSchemaName())
+                    .withName(from.getTableName());
+            com.amazonaws.services.glue.model.Table table = stats.getGetTable().call(() -> glueClient.getTable(getTableRequest).getTable());
+            TableInput tableInput = getTableInput(
+                    to.getTableName(),
+                    Optional.ofNullable(table.getOwner()),
+                    table.getParameters());
+            CreateTableRequest createTableRequest = new CreateTableRequest()
+                    .withDatabaseName(to.getSchemaName())
+                    .withTableInput(tableInput);
+            stats.getCreateTable().call(() -> glueClient.createTable(createTableRequest));
+            newTableCreated = true;
+            deleteTable(from.getSchemaName(), from.getTableName());
+        }
+        catch (Exception e) {
```

Comment:
Avoid catching `Exception` since it can hide `InterruptedException`. Instead, catch `RuntimeException` and any necessary checked exceptions.